runelite-client/src/main/java/net/runelite/client/plugins/npcunaggroarea/NpcAggroAreaConfig.java

```diff
@@ -0,0 +1,106 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.npcunaggroarea;
+
+import java.awt.Color;
+import net.runelite.client.config.Config;
+import net.runelite.client.config.ConfigGroup;
+import net.runelite.client.config.ConfigItem;
+
+@ConfigGroup("npcUnaggroArea")
+public interface NpcAggroAreaConfig extends Config
+{
+	String CONFIG_GROUP = "npcunaggroarea";
+	String CONFIG_CENTER1 = "center1";
+	String CONFIG_CENTER2 = "center2";
+	String CONFIG_LOCATION = "location";
+	String CONFIG_DURATION = "duration";
+
+	@ConfigItem(
+		keyName = "npcUnaggroAlwaysActive",
+		name = "Always active",
+		description = "Always show this plugins overlays<br>Otherwise, they will only be shown when any NPC name matches the list",
+		position = 1
+	)
+	default boolean alwaysActive()
+	{
+		return false;
+	}
+
+	@ConfigItem(
+		keyName = "npcUnaggroNames",
+		name = "NPC names",
+		description = "Enter names of NPCs where you wish to use this plugin",
+		position = 2
+	)
+	default String npcNamePatterns()
+	{
+		return "";
+	}
+
+	@ConfigItem(
+		keyName = "npcUnaggroShowTimer",
+		name = "Show timer",
+		description = "Display a timer until NPCs become unaggressive",
+		position = 3
+	)
+	default boolean showTimer()
+	{
+		return true;
+	}
+
+	@ConfigItem(
+		keyName = "npcUnaggroShowAreaLines",
+		name = "Show area lines",
+		description = "Display lines, when walked past, the unaggressive timer resets",
+		position = 4
+	)
+	default boolean showAreaLines()
+	{
+		return true;
+	}
+
+	@ConfigItem(
+		keyName = "npcUnaggroCollisionDetection",
+		name = "Collision detection",
+		description = "Only show lines where they can be walked through",
+		position = 5
+	)
+	default boolean collisionDetection()
```

Comment:
Does this even need config option?

Reply:
What should we do if we remove the config option? Keep collision or not? I kinda like having both options, no collision detection makes it easier to see exactly where the area is, but adding collision detection fixes issues where the lines goes off the map like this
![image](https://user-images.githubusercontent.com/26200523/47398062-93a1ef00-d732-11e8-9bae-dea327744dd9.png)

Reply:
Keep collision. If you cant walk to end of the area it is not important to know full area imo (as that is mostly just debug info and not actually useful info)